src/devices/Pca9685/Pca9685.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.I2c;
+using System.Threading;
+using static Iot.Device.Pca9685.Register;
+using static Iot.Device.Pca9685.Mode1;
+using static Iot.Device.Pca9685.Mode2;
+
+namespace Iot.Device.Pca9685
+{
+    /// <summary>
+    /// PCA9685 PWM LED/servo controller
+    /// </summary>
+    public class Pca9685 : IDisposable
+    {
+
+        /// <summary>
+        /// I2C Device
+        /// </summary>
+        private I2cDevice _device;
+
+        /// <summary>
+        /// Get default clock rate. Set if you are using external clock.
+        /// </summary>
+        public double ClockRate { get; set; } = 25000000;   // 25MHz
+
+        /// <summary>
+        /// Set PWM frequency or get effective value.
+        /// </summary>
+        public double PwmFrequency
+        {
+            get => _pwmFrequency;
+            set
+            {
+                Prescale = GetPrescale(value);
+            }
+        }
+
+        /// <summary>
+        /// Set PWM frequency using prescale value or get the value.
+        /// </summary>
+        public byte Prescale
```

Comment:
I think this could be internal since you can already do it with the other property - personally find it a bit confusing but fine to leave if you think it adds enough value

Reply:
PCA9685's PWM frequency is directly linked to `Prescale`. Users can change it or change `Frequency`(indirectly, but calculating creates deviation). If actual frequency is different from users want, they can know why from `Prescale`.